game-core/src/main/java/games/strategy/triplea/ResourceLocationTracker.java

```diff
@@ -10,42 +15,30 @@
  */
 @UtilityClass
 class ResourceLocationTracker {
+  // All maps must have at least a "baseTiles" folder
+  private static final String REQUIRED_ASSET_EXAMPLE_FOLDER = "baseTiles/";
 
   /**
-   * master zip is the zipped folder format you get when downloading from a map repo via the 'clone
-   * or download' button.
-   */
-  static final String MASTER_ZIP_MAGIC_PREFIX = "-master/map/";
-
-  static final String MASTER_ZIP_IDENTIFYING_SUFFIX = "-master.zip";
-
-  /**
-   * * Will return an empty string unless a special prefix is needed, in which case that prefix is *
-   * constructed based on the map name. * *
-   *
-   * <p>The 'mapPrefix' is the path within a map zip file where we will then find any map contents.
-   * * For example, if the map prefix is "map", then when we expand the map zip, we would expect *
-   * "/map" to be the first folder we see, and we would expect things like "/map/game" and *
-   * "/map/polygons.txt" to exist.
+   * Will return an empty string unless a special prefix is needed, in which case that prefix is *
+   * constructed based on where the {@code baseTiles} folder is located within the zip.
    *
-   * @param mapName Used to construct any special resource loading path prefixes, used as needed
-   *     depending upon which resources are in the path
    * @param resourcePaths The list of paths used for a map as resources. From this we can determine
    *     if the map is being loaded from a zip or a directory, and if zip, if it matches any
    *     particular naming.
    */
-  static String getMapPrefix(final String mapName, final URL[] resourcePaths) {
-    final boolean isUsingMasterZip =
-        Arrays.stream(resourcePaths)
-            .map(Object::toString)
-            .anyMatch(path -> path.endsWith(MASTER_ZIP_IDENTIFYING_SUFFIX));
-
-    // map skins will have the full path name as their map name.
-    if (mapName.endsWith("-master.zip")) {
-      return mapName.substring(0, mapName.length() - "-master.zip".length())
-          + MASTER_ZIP_MAGIC_PREFIX;
-    } else {
-      return isUsingMasterZip ? mapName + MASTER_ZIP_MAGIC_PREFIX : "";
+  static String getMapPrefix(final URL[] resourcePaths) {
+    for (final URL url : resourcePaths) {
+      try (ZipFile zip = new ZipFile(new File(url.toURI()))) {
+        final Optional<? extends ZipEntry> e =
```

Comment:
nit, the single letter variable names can be hard to decipher. Would you mind replacing it with something more descriptive, like zipFile, or fileInZip, or zipFileEntry?

Similarly for the `$` in the lambda.

Reply:
I like using single character variable names to communicate to the reader it's not an "important" variable- very often in lambdas or for loops. I'm telling you as the reader to "keep going" and focus on other parts of the code. But I'm happy to conform to the current conventions in the code.

Reply:
@DanVanAtta I just pushed another commit that removes the single char variable names